tests/ha/control_pod/test_control_pod_restart.py

```diff
@@ -910,33 +907,31 @@ def test_iam_bkt_cruds_during_ctrl_pod_failover(self):
         LOGGER.info("Waiting for %s sec for background operations to start",
                     HA_CFG["common_params"]["10sec_delay"])
         time.sleep(HA_CFG["common_params"]["10sec_delay"])
-
-        LOGGER.info("Control pod %s is hosted on %s node", self.control_pod_name, self.control_node)
-        failover_node = self.system_random.choice([ele for ele in self.host_worker_list if ele !=
-                                                   self.control_node])
-        LOGGER.debug("Fail over node is: %s", failover_node)
-        event.set()
-        LOGGER.info("Step 3: Failover control pod %s to node %s and check cluster status",
-                    self.control_pod_name, failover_node)
-        pod_yaml = {self.control_pod_name: self.modified_yaml}
-        resp = self.ha_obj.failover_pod(pod_obj=self.node_master_list[0], pod_yaml=pod_yaml,
-                                        failover_node=failover_node)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 3: Successfully failed over control pod to %s. Cluster is in good state",
-                    failover_node)
+        LOGGER.info("Step 3: Restart %s control pods while creation/deletion of IAM users and "
+                    "buckets is running in background.", self.repl_num - 1)
+        for count in range(HA_CFG["common_params"]["short_loop"]):
+            LOGGER.info("Restarting %s control pods for loop : %s", self.repl_num - 1, count + 1)
+            ctrl_pod_list = self.system_random.sample(pod_list, self.repl_num - 1)
+            for pod in ctrl_pod_list:
+                resp = self.node_master_list[0].delete_pod(pod)
+                assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3.1: Check cluster status")
+            delay = HA_CFG["common_params"]["60sec_delay"]
+            resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=delay)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3.1: Cluster status is online")
+        LOGGER.info("%s pod are restarted successfully in loop of %s", self.repl_num - 1,
+                    HA_CFG["common_params"]["short_loop"])
         event.clear()
-
-        self.restore_pod = self.deploy = True
-        LOGGER.info("Step 4: Verify if IAM users %s are persistent across control pod failover",
+        LOGGER.info("Step 4: Verify if IAM users %s are persistent across control pods restart",
```

Comment:
Should we also check that all background operations are successful as 1 control pod will always be alive?

Reply:
After looking at this, I guess originally we had written this step even before joining other threads. Should we then shift this afterwards after all background activity is done? So all validation will be in one go?